### DISCUSSION TITLE
Query on Textual

### DISCUSSION THREAD

**Discussion by never00rei** [Textualize/textual#110] [General]
In regards to textual, will you be able to utilise the curses module inside of a widget?

**Comment by willmcgugan** [comment]
Highly unlikely. But I hope that Textual will eventually have enough widgets that you would never need to.

Did you have a use case in mind?

**Reply by never00rei** [reply]
Yeah, so I'm building a small concept IRC like chat client. Loosely themed around the Matrix. The idea is use textual to draw the console, split into three areas:
1. Operators console, as seen in the matrix - where they can see the encoded construct. I found a few tutorials for "matrix rain" but they all use curses to draw on the terminal itself. It semi-works - I can push the widget, but curses prints over everything being drawn by the widget like borders ETC.
2. Chat window for chat history.
3. Input window - for the user to reply with.

**Reply by tusharsadhwani** [reply]
Making a matrix rain widget sounds like a fun side project
